src/resty/ljsonschema/init.lua

```diff
@@ -877,7 +877,20 @@ generate_validator = function(ctx, schema)
       ctx:stmt(sformat('  %s(%s)', validator, ctx:param(1)), op)
     end
     ctx:stmt(') then')
-    ctx:stmt('  return false, "object matches none of the alternatives"')
+    ctx:stmt('  local unmatched = ""')
+    ctx:stmt('  do')
+    for i, subschema in ipairs(schema.anyOf) do
+      ctx:stmt(        '  local was_matched')
+      ctx:stmt(        '  local error_message')
+      local validator = ctx:validator({ 'anyOf', tostring(i-1) }, subschema)
+      ctx:stmt(sformat('  was_matched, error_message = %s(%s)', validator, ctx:param(1)))
+      ctx:stmt(        '  if not was_matched then')
+      ctx:stmt(        '    unmatched = unmatched .. \'\\n\\t - \' .. error_message')
+      ctx:stmt(        '  end')
+    end
+    ctx:stmt('  end')
+    ctx:stmt(sformat('  return false, %s("object needs one of the following rectifications: %%s", unmatched)',
+                     ctx:libfunc('string.format')))
```

Comment:
can't we record the error messages in the loop above, and then just concat them when the whole thing fails, instead of looping again?

I think this change also warrants a test to validate it.

Reply:
I imagine I'd remove that previous loop and then just use this section, actually.

I'll change and test this in a couple of weeks when I have some time!